### PR TITLE
Adding Dnsnamelable and fqdn(readonly) properties to container group in 2018-02-01-preview API

### PR DESCRIPTION
**Pull request by xizha162**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ X] The title of the PR is clear and informative.
- [ X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X ] If applicable, the PR references the bug/issue that it fixes.
- [ X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ X] My spec meets the review criteria:
  - [X ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by xizhamsft** [comment]
@zhedahht Could you review this PR?

**Review by zhedahht — APPROVED** [review]

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/examples/ContainerLogsList.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "parameters": {
+    "subscriptionId": "subid",
+    "api-version": "2018-02-01-preview",
+    "resourceGroupName": "demo",
+    "containerGroupName": "demo1",
+    "containerName": "container1",
+    "tail": "10"
```
Comment (jianghaolu) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/examples/ContainerLogsList.json]: The type of tail is integer.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json [hunk 2]
```diff
@@ -0,0 +1,1099 @@
...
+          "description": "The path within the container where the volume should be mounted. Must not contain colon (:).",
+          "type": "string"
+        },
+        "readOnly": {
+          "description": "The flag indicating whether the volume mount is read-only.",
+          "type": "boolean"
+        }
+      }
+    },
+    "ContainerGroup": {
```
Comment (jianghaolu) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: Tracked resource 'ContainerGroup' must have patch operation that at least supports the update of tags. It's strongly recommended that the PATCH operation supports update of all mutable properties as well.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json [hunk 3]
```diff
@@ -0,0 +1,1099 @@
...
+                "osType": {
+                  "type": "string",
+                  "description": "The operating system type required by the containers in the container group.",
+                  "enum": [
+                    "Windows",
+                    "Linux"
+                  ],
+                  "x-ms-enum": {
+                    "name": "OperatingSystemTypes",
+                    "modelAsString": true
```
Comment (jianghaolu) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: Will there be more OS support added?

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json (hunk 3 above)
Comment (xizhamsft) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: For now, no.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json (hunk 2 above)
Comment (xizhamsft) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/containerInstance.json]: There is another PR for patch operation. I combined the changes from that PR to this one.

**Comment by xizhamsft** [comment]
@jianghaolu  Also changed the tail value to integer per your comment.

**Comment by xizhamsft** [comment]
@jianghaolu  Addressed all comments. Please take a look.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/examples/ContainerLogsList.json (hunk 1 above)
Comment (xizhamsft) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/examples/ContainerLogsList.json]: fixed. THanks.

**Comment by vivsriaus** [comment]
@ravbhatnagar to review API changes

**Comment by ravbhatnagar** [comment]
Looks good. Only thing you should for all APis is  a default response for all apis to capture the errors. Check example in Batch RP swagger. Not blocking PR merge for this. But something you should add if you can now, or in the next rev.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/examples/ContainerGroupsUpdate.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        "tags": {
+          "tag1key": "tag1Value",
+          "tag2key": "tag2Value"
+        }
+      }
+    },
+    "responses": {
+      "200": {
+        "body": {
+          "id": "/subscriptions/subid/resourceGroups/demoResource/providers/Microsoft.ContainerInstance/containerGroups/demo1",
```
Comment (jianghaolu) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/examples/ContainerGroupsUpdate.json]: Sorry I just caught this now - but the `properties` property bag defined in `ContainerGroup` is missing in the response.

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/examples/ContainerGroupsUpdate.json (hunk 4 above)
Comment (xizhamsft) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2018-02-01-preview/examples/ContainerGroupsUpdate.json]: @jianghaolu fixed. Please take a look.

**Review by jianghaolu — APPROVED** [review]

**Pull request merged**